### PR TITLE
[DBotAverageScore] Migrate to Python, and fix '0' average causing a null value

### PR DESCRIPTION
**Pull request by MichaelYochpaz**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-26026)

## Description
Fix an issue where if all reputation values are '0', a null value is added to context.
Also migrate to Python, and update the logic to ignore '0' values in the calculation (after consulting @Ni-Knight).

## Screenshots
Output before changes:
Screenshot 2023-07-10 at 13 46 51

Output after changes:
Screenshot 2023-07-10 at 13 46 59

### CODE REVIEW

**Comment by MichaelYochpaz** [comment]
@DeanArbel adding you since we'll need a force-merge, as a validation fails for removing the previous JS file.

File: Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    return CommandResults(
+        outputs_prefix='DBotAvgScore',
+        outputs_key_field='Indicator',
+        outputs=context_output,
+        readable_output=tableToMarkdown('DBot Average Scores', t=context_output),
+    )
+
+
+def calculate_average_score(indicator: str, scores_list: list[int]) -> dict:
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.py:43]: I would place this function first in the file

File: Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    If no scores are provided, the average is 0.
+
+    Args:
+        indicator (str): The indicator for which the average is calculated.
+        scores_list (list[int]): A list of scores.
+
+    Returns:
+        dict: A context entry for a single indicator, containing the average score.
+    """
+    scores_list = [score for score in scores_list if score]  # Remove '0' values
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.py:56]:
```suggestion
    scores_list = [score for score in scores_list if score != 0]
```
We want to remove only '0' values, not null 
It is better to not rely on the truthiness of the score value.

**Review by bziser — COMMENTED** [review]
great

**Review by Ni-Knight — APPROVED** [review]

**Review by bziser — APPROVED** [review]

File: Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.yml [hunk 3]
```diff
@@ -3,17 +3,18 @@ commonfields:
...
-type: javascript
-comment: Calculates average score for each indicator from context
+type: python
+comment: The script calculates the average DBot score for each indicator in the context.
 tags:
 - Utility
 outputs:
 - contextPath: DBotAvgScore.Indicator
-  description: The indicator we calculate average score for
+  description: The indicator the average score is for
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.yml:12]:
```suggestion
  description: The indicator the average score is for.
```

File: Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.yml [hunk 4]
```diff
@@ -3,17 +3,18 @@ commonfields:
...
 - contextPath: DBotAvgScore.Score
-  description: The average score per indicator
+  description: The average score for the indicator
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/DBotAverageScore/DBotAverageScore.yml:14]:
```suggestion
  description: The average score for the indicator.
```

File: Packs/CommonScripts/Scripts/DBotAverageScore/README.md [hunk 5]
```diff
@@ -1,22 +1,29 @@
...
 
 ## Outputs
+
 ---
 
 | **Path** | **Description** | **Type** |
 | --- | --- | --- |
-| DBotAvgScore.Indicator | The indicator that the average score was calculated for. | Unknown |
-| DBotAvgScore.Score | The average score per indicator. | Unknown |
+| DBotAvgScore.Indicator | The indicator the average score is for | Unknown |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/DBotAverageScore/README.md:28]:
```suggestion
| DBotAvgScore.Indicator | The indicator the average score is for, | Unknown |
```

File: Packs/CommonScripts/Scripts/DBotAverageScore/README.md [hunk 6]
```diff
@@ -1,22 +1,29 @@
...
+| DBotAvgScore.Score | The average score for the indicator | Unknown |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/DBotAverageScore/README.md:29]:
```suggestion
| DBotAvgScore.Score | The average score for the indicator, | Unknown |
```

**Comment by ShirleyDenkberg** [comment]
@Ni-Knight @bziser @DeanArbel Doc review completed.

**Pull request merged**